src/function_classes/sparse_parity.py

```diff
@@ -0,0 +1,26 @@
+import torch
+import torch.distributions as D
+
+from typing import Any
+
+from utils import RandomPermutationDistribution
+from core import FunctionClass
+
+class SparseParityRegression(FunctionClass):
+
+    def __init__(self, k: int = 2, *args: Any, **kwargs: Any):
+        self.k = k
+        super(SparseParityRegression, self).__init__(*args, **kwargs)
+
+    def _init_param_dist(self) -> D.Distribution:
+        return RandomPermutationDistribution(self.k, self.x_dim)
```

Comment:
The batch size isn't passed here ... you may be sampling only one function per batch

Reply:
true, right now one set of indicies is chosen per prompt in the batch